### PR TITLE
Replace fastutil dependency with GTNHLib

### PR DESCRIPTION
**Pull request by tth05**
The newest fastutil version now ships with GTNHLib

### CODE REVIEW

File: dependencies.gradle [hunk 1]
```diff
@@ -56,7 +56,7 @@ dependencies {
 
     // Iris Shaders
     compileOnly('org.jetbrains:annotations:24.0.1')
-    shadowImplementation("it.unimi.dsi:fastutil:8.2.1") // Apache 2.0
+    api("com.github.GTNewHorizons:GTNHLib:0.2.1:dev")
```
Comment (mitchej123) [inline dependencies.gradle:59]: Can you also add gtnhlib to AngelicaMod's dependencies?

Comment (mitchej123) [reply dependencies.gradle:59]: And document the requirement in the readme.

File: src/main/java/com/gtnewhorizons/angelica/AngelicaMod.java [hunk 2]
```diff
@@ -15,7 +15,7 @@
         name = "Angelica",
         version = Tags.VERSION,
         dependencies = " before:lwjgl3ify@[1.5.3,);" + " after:hodgepodge@[2.4.4,);" + " after:CodeChickenCore@[1.2.0,);"
-                     + " after:archaicfix@[0.7.0,);",
+                     + " after:archaicfix@[0.7.0,);" + " after:gtnhlib@[0.2.1,);",
```
Comment (mitchej123) [inline src/main/java/com/gtnewhorizons/angelica/AngelicaMod.java:18]: this should be a `required-after`; the other mods are optional and if loaded needed to be that version; angelica now won't function without NH lib

Comment (mitchej123) [reply src/main/java/com/gtnewhorizons/angelica/AngelicaMod.java:18]:
```suggestion
                     + " after:archaicfix@[0.7.0,);" + " required-after:gtnhlib@[0.2.1,);",
```

Comment (tth05) [reply src/main/java/com/gtnewhorizons/angelica/AngelicaMod.java]: Yeah makes sense. I'm not familiar with the dependency string format

**Pull request merged**